### PR TITLE
App Menus

### PR DESCRIPTION
**Pull request by fhenrich33**
Desktop/Tablet:

Screen Shot 2024-11-07 at 19 18 35

Screen Shot 2024-11-07 at 19 18 41

Screen Shot 2024-11-07 at 19 18 46

firefox_ZUdfbmvCT6

Mobile:

firefox_Ow7ExjCcMm

### CODE REVIEW

**Comment by fhenrich33** [comment]
[2.0 FE Full Application Menus for Mobile and Desktop #large](https://trello.com/c/K9DpHcS9/1513-20-fe-full-application-menus-for-mobile-and-desktop-large)

**Comment by HaGuesto** [comment]
@aerinsol could you check out the screenshots from Felipe if it is the way you like it. 

FYI: I do not really get why we switch to a top menu for large desktops. There we would have the most real estate to show the drawer menu.

Also is the drawer menu on small desktop/tablet collapsable, meant to be hidden?

**Comment by aerinsol** [comment]
> @aerinsol could you check out the screenshots from Felipe if it is the way you like it. 
> 
> FYI: I do not really get why we switch to a top menu for large desktops. There we would have the most real estate to show the drawer menu.
> 
> Also is the drawer menu on small desktop/tablet collapsable, meant to be hidden?

Two reasons. 1. the last time we put this design forward, it was said the drawer menu was too difficult to implement. 2. Sometimes there isn't enough vertical real estate to show things. We see this on small screens for Dropapp.

File: front/src/components/HeaderMenu/HeaderMenuContainer.tsx [hunk 1]
```diff
@@ -77,6 +89,84 @@ function HeaderMenuContainer() {
           },
         ],
       },
+      {
+        text: "Beneficiares",
+        requiredAbp: ["manage_beneficiaries"], // TODO: this or "view_beneficiaries"? "view_beneficiaries" doesn't seem to work.
```
Comment (HaGuesto) [inline front/src/components/HeaderMenu/HeaderMenuContainer.tsx:94]: change this to create_beneficiaries

Comment (HaGuesto) [reply front/src/components/HeaderMenu/HeaderMenuContainer.tsx:94]: view_beneficiary does not exist as ABP

File: front/src/components/HeaderMenu/HeaderMenuContainer.tsx [hunk 2]
```diff
@@ -77,6 +89,84 @@ function HeaderMenuContainer() {
...
+            link: `${oldAppUrlWithBase}&action=people`,
+            name: "Manage Beneficiares",
+            requiredAbp: ["manage_beneficiaries"],
+            external: true,
+          },
+        ],
+      },
+      {
+        text: "Free Shop",
+        requiredAbp: [], // TODO: revisit this. See https://trello.com/c/5IIOrOmY#comment-6720d286c59afc434a52e302
```
Comment (HaGuesto) [inline front/src/components/HeaderMenu/HeaderMenuContainer.tsx:112]: pls add here checkout_beneficiaries and remove the TODO

File: front/src/components/HeaderMenu/HeaderMenuContainer.tsx [hunk 3]
```diff
@@ -77,6 +89,84 @@ function HeaderMenuContainer() {
...
+            link: `${oldAppUrlWithBase}&action=give2all`,
+            name: "Give Tokens To All",
+            requiredAbp: ["manage_tokens"],
+            external: true,
+          },
+        ],
+      },
+      {
+        text: "Admin",
+        requiredAbp: [],
```
Comment (HaGuesto) [inline front/src/components/HeaderMenu/HeaderMenuContainer.tsx:142]: add here manage_volunteer

File: front/src/components/HeaderMenu/HeaderMenuContainer.tsx [hunk 4]
```diff
@@ -77,6 +89,84 @@ function HeaderMenuContainer() {
...
+          {
+            link: `${oldAppUrlWithBase}&action=locations`,
+            name: "Edit Warehouses",
+            requiredAbp: ["manage_warehouses"],
+            external: true,
+          },
+          {
+            link: `${oldAppUrlWithBase}&action=cms_users`,
+            name: "Manage Users",
+            requiredAbp: ["cms_users"],
```
Comment (HaGuesto) [inline front/src/components/HeaderMenu/HeaderMenuContainer.tsx:165]: change here to manage_volunteer. cms_users does not exist as ABP

File: front/src/components/HeaderMenu/HeaderMenuDesktop.tsx [hunk 5]
```diff
@@ -1,141 +0,0 @@
...
-        {props.links.map((subMenu) => (
-          <MenuItem
-            py={2}
-            px={3}
-            key={subMenu.name.replace(/\s+/g, "-").toLowerCase()}
-            as={NavLink}
-            to={subMenu.link}
-            reloadDocument={subMenu.external}
-          >
-            {subMenu.name}&nbsp;{subMenu.beta && <sup>beta</sup>}
```
Comment (HaGuesto) [inline front/src/components/HeaderMenu/HeaderMenuDesktop.tsx:85]: @fhenrich33 pls reimplement the `<sup>beta</sup>` in your menues

**Review by HaGuesto — APPROVED** [review]
# code review
- a few permissions need to changed
- I'd like to see some tests (at least that the ABPs are checked correctly and that the links are right)


# functional review
## general 
[for @aerinsol to decide]
I'd like to have a consistent design on how the MainMenues and subMenues look for all three screensizes. At the moment, some are gray, some are white, some have a border, some don't, some are completly filled with the background, some have a margin.
I'd suggest to 
- make the main menues white and the submenues gray. 
- remove the black border and replace with a slight gray as used for separation of accordions
- not have any margins between the buttons.

## mobile menu
- the vertical padding between HeadMenu and SubMenu should be aligned.


## tablet view
- the content overflows over/under the drawer menu if you are moving to the right.
- [for @aerinsol to decide] the black border is a bit too "in your face" for my taste: I'd remove the border from the top, left, and bottom and align color and thickness with the separation lines of the accordion (slight gray and thinner)
Screenshot_2024-10-31_15-33-37


## top menu
- [for @aerinsol to decide] The main menues have a gray background and look like action buttons on the table. I also find the black border aesthetically not really pleasing. I'd suggest to make them white and remove the border or remove the whole top menu and change it to the drawer for large screensizes.
Screenshot_2024-10-31_15-39-12

PS. @fhenrich33 I do not need to review this again. Good job. Pls do not merge before tests are in.

**Comment by HaGuesto** [comment]
@aerinsol could you tell me what you think of my suggestions on the design mentioned in the review? I marked them with [@aerinsol to decide]

**Comment by aerinsol** [comment]
Hmm. I had a think about this, and I think keeping the drawer and the mobile menu only, and dropping the top menu is the best solution for consistency. The drawer menu is the most understandable for someone not used to tech and has the best information hierarchy. 

Yes, I would like the drawer menu to be collapsible (close able in Chakra) and also to have a drop shadow over the main screen (@fhenrich33 you can just pick another component with a drop shadow, if you can't find it we can leave it for a follow up).

I'll go through the options now to see how to align those two.

**Comment by aerinsol** [comment]
Also @fhenrich33 good job selecting all the correct icons! I know it's a small thing but I spent a lot of time trying to sort through all of them and label them in Figma correctly 😅

**Comment by aerinsol** [comment]
> 1. align the visual design of **main menu buttons** in all screen sizes

Yes, let's match the drawer color scheme for both menus. White bg for main menu buttons, grey bg for child menu buttons, and red when clicked on or expanded. Font is black except when the background is red, then it's white and bolded (@fhenrich33 you are bolding it `onClick` in the drawer menu, right? Keep it to whatever you're doing there.) 


> - there should be no spacing/margin between the main menu buttons (this is currently only the case in the mobile hamburger menu). Instead we could add the slight gray separation lines from the accordion.

So actually, there are supposed to be 3 separate groups in the mobile hamburger menu with `<MenuDivider />` separating them. Possibly impossible to do in Figma. Divisions should be at Scan QR Label | Statistics ... Admin | Account

Also,@fhenrich33, Scan QR Label Menu button should be the same size as all the others, and "Beneficiaries" is spelled incorrectly in all menus. 

> 
> 3. align the drawer menu border in tablet screen size with the standard design of the accordions
> At the moment the border of the drawer menu is black and fully surrounding the drawer menu. 
> I'd 
> - remove the border from the top, left, and bottom so that the border just becomes a separation line between left and right. 
> - use the color and thickness of the accordion separation lines to make the border less visually outstanding.

This is fine, as I said earlier I think a drop shadow would be helpful but not a deal breaker for production.

**Comment by aerinsol** [comment]
As discussed in [Slack](https://boxwise.slack.com/archives/CB1UYDGLU/p1730649109027969) and other places, the icon and text for Admin will be adjusted to a lock icon and the text "Coordinator Admin"; Logout will also now have a key icon. 

Reason for icon adjustment: It is intended to better mimics offline reality for non-tech people. Something with a lock mimics a locked door or cabinet that only certain people have access to; logging out is similar to taking a key to lock up a shop or a warehouse at the end of the day.

File: front/src/components/HeaderMenu/HeaderMenuContainer.tsx [hunk 6]
```diff
@@ -139,7 +139,7 @@ function HeaderMenuContainer() {
       },
       {
         text: "Coordinator Admin",
-        requiredAbp: [],
+        requiredAbp: ["manage_volunteer"],
```
Comment (pylipp) [inline front/src/components/HeaderMenu/HeaderMenuContainer.tsx:142]: This needs an extra s: `manage_volunteers`
ABPs are listed [here](https://docs.google.com/spreadsheets/d/1W4YWcc59wUFUWgReumdH6DQ4zU7JcTgvf6WEbdqaGHQ/edit?gid=1124938013#gid=1124938013) (4th sheet).

Comment (fhenrich33) [reply front/src/components/HeaderMenu/HeaderMenuContainer.tsx]: Thank you @pylipp !

**Comment by HaGuesto** [comment]
@fhenrich33 Thanks for the update! I'm quite happy with it.
One feedback:
I generally prefer doing rather global definitions. I'd for example would like to define the screensizes at some point globally or the Menu component instead of inside each component.


@aerinsol There are a few pointers for you to judge since it is not 100% clear to me from the figma:
- the behavior of the expanding menu on the large screen should be on topo of the content or the content should be squeezed to the right?
- the account and logout button is still on the top right
- the <MenuDivider /> is implemented in mobile screen but not on the large screen
- should the accordion in the menu stay open or should always only one accordion be open?

[I did a quick screen recording for you to judge](https://www.loom.com/share/86d1a08bfaaf4cce873fcd289d230aa1?sid=6f01f2ae-72a6-4e34-8614-5f88ad56b452)

**Comment by aerinsol** [comment]
@HaGuesto 

> * the behavior of the expanding menu on the large screen should be on topo of the content or the content should be squeezed to the right?

It should be squeezed to the right, and the menu should be open by default. We should probably use arrows to show open / closed - I can add that if needed.

> * the account and logout button is still on the top right

Yeah, I'll probably change this in the next iteration. I thought the top right might be nicer but now I see it's probably not. 

> * the  is implemented in mobile screen but not on the large screen

? 

> * should the accordion in the menu stay open or should always only one accordion be open?

Only one accordion should be open on the mobile menu at a given time.

**Comment by fhenrich33** [comment]
@aerinsol 

> It should be squeezed to the right, and the menu should be open by default. We should probably use arrows to show open / closed - I can add that if needed.

Chakra doesn't support that out of the box for Drawers, we would need to roll our own implementation. See https://v2.chakra-ui.com/docs/components/drawer

> Only one accordion should be open on the mobile menu at a given time.

Fixed that right after @HaGuesto comment.

**Comment by aerinsol** [comment]
> @aerinsol 
> 
> > It should be squeezed to the right, and the menu should be open by default. We should probably use arrows to show open / closed - I can add that if needed.
> 
> Chakra doesn't support squeezing out of the box for Drawers, we would need to roll our own implementation. 
> 

It looks like it does, but only if you purchase Chakra Pro.

https://pro.chakra-ui.com/components/application/sidebars

If that's the case, for the sake of time I am happy to go back to the fixed sidebar style you originally put out @fhenrich33

**Comment by fhenrich33** [comment]
>
> It looks like it does, but only if you purchase Chakra Pro.
> 
> https://pro.chakra-ui.com/components/application/sidebars

Seems to be only for version 3 onwards. Reverted to fixed!

**Pull request merged**